storage/objects.go

```diff
@@ -258,6 +260,21 @@ type PutObjectInput struct {
 func (s *ObjectsClient) Put(ctx context.Context, input *PutObjectInput) error {
 	fullPath := path.Clean(path.Join("/", s.client.AccountName, input.ObjectPath))
 
+	err := checkDirectoryTreeExists(*s, ctx, input.ObjectPath)
+	if err != nil {
+		if strings.Contains(err.Error(), "ResourceNotFound") { //ResourceNotFound when dir.List doesn't exist
```

Comment:
I don't normally like this given a type system. We discussed error handling yesterday though so I'm fine with this for now.

Reply:
Yeah error handling needs to change all over the shop!

Reply:
We have removed at least some of the string matching for now